MediaBrowser.Api/Library/LibraryService.cs

```diff
@@ -759,13 +759,14 @@ private void LogDownload(BaseItem item, User user, AuthorizationInfo auth)
         {
             try
             {
-                _activityManager.Create(new ActivityLogEntry
+                _activityManager.Create(new Jellyfin.Data.Entities.ActivityLog(
+                    string.Format(_localization.GetLocalizedString("UserDownloadingItemWithValues"), user.Name, item.Name),
+                    "UserDownloadingContent",
+                    auth.UserId,
+                    DateTime.UtcNow,
+                    LogLevel.Trace)
```

Comment:
So this is actually something I was thinking about: The auto-generated code requires the non-null columns to be given in the constructor, while the old code left the responsibility to the caller to populate those fields. I think the latter approach looks much cleaner, at the cost of being slightly less safe.

Reply:
Speaking up here, I feel higher safety is important. Ideally we never have to worry, but more safety means less burden on reviewers when community members send in PRs.

Anything we can do to stop randos from making a mess of things should be given a lot of weight imo.

Reply:
Yeah, I would also prefer to keep additional safety. As a compromise, we could add the non-required properties as optional parameters to the constructor

Reply:
Yeah. Anything we can do to prevent extra review load and a new potential source of bugs from popping up.

Compiler should be our friend, the more we can ask it to do the better imo.